lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm

```diff
@@ -152,6 +152,16 @@ sub serialize_aliases {
 
     return unless defined $inc && $inc->aliases;
 
+    # We don't show aliases again for recording artists if they're on the release or track AC
+    if ($entity->isa('MusicBrainz::Server::Entity::Artist')) {
+        if (my $release_ac = $stash->{release_artist_credit}) {
+            if (my $track_ac = $stash->{track_artist_credit}) {
+                return if (grep { $_->artist_id == $entity->id } $release_ac->all_names);
+                return if (grep { $_->artist_id == $entity->id } $track_ac->all_names);
+            }
+        }
+    }
```

Comment:
Can we copy the same code from https://github.com/metabrainz/musicbrainz-server/blob/6b9a6446d6ed6593ffe23e39c2df29154680771c/lib/MusicBrainz/Server/WebService/Serializer/JSON/2/Utils.pm#L284-L291 here? Or factor it out into a `is_artist_used_by_track_or_release` or something?

(Not actually sure there's a case that matters for aliases where `release_artist_credit` is defined and not `track_artist_credit`. But I might be forgetting something. It'd be nice to have the code shared in any case.)

Reply:
I specifically based this on that but changed how it works, so that it must ensure a  track_artist_credit exists. Otherwise, unless I'm getting stuff wrong, this would also apply to track artist credits and actually break stuff that is there now.

Reply:
Got it, thanks. Can you add a small line to the comment above saying you check for both so that it doesn't apply to tracks to maintain compatibility?